samples/compression/compression-custom/main.cpp

```diff
@@ -0,0 +1,261 @@
+
+private:
+    static const MetaString id;
+};
+
+const MetaString MyLZWCompressor::id = "my_lzw";
+
+
+int main(int argc, char** argv)
+{
+    if (argc != 2)
+    {
+        std::cerr << "Error: Wrong number of arguments." << std::endl;
+        std::cerr << "Usage: " << argv[0] << " <video_file> " << std::endl;
+        return -1;
+    }
+
+    vmf::Log::setVerbosityLevel(vmf::LOG_NO_MESSAGE);
+    vmf::initialize();
+
+    vmf::vmf_string videoFile(argv[1]);
+
+    vmf::MetadataStream mdStream;
+    // Open metadata stream
+    if (!mdStream.open(videoFile, vmf::MetadataStream::ReadWrite))
+    {
+        std::cerr << "Can't open file '" + videoFile + "'" << std::endl;
+        exit(1);
+    }
+
+    const string GPS_DESC = "gps";
+    const string GPS_COORD_LAT_FIELD = "lat";
+    const string GPS_COORD_LNG_FIELD = "lng";
+    const string GPS_TIME_FIELD = "Time";
+    const string GPS_SCHEMA_NAME = "gps_coords_schema";
+
+    // Create a GPS metadata field descriptions
+    vector<FieldDesc> fieldDesc;
+
+    fieldDesc.push_back(FieldDesc(GPS_COORD_LAT_FIELD, Variant::type_real)); // GPS coordinates
+    fieldDesc.push_back(FieldDesc(GPS_COORD_LNG_FIELD, Variant::type_real)); // GPS coordinates
+    fieldDesc.push_back(FieldDesc(GPS_TIME_FIELD,      Variant::type_real)); // Associated time in msecs
+
+    // Create GPS metadata description
+    shared_ptr<MetadataDesc> gpsDesc(new MetadataDesc(GPS_DESC, fieldDesc));
+
+    // Create GPS metadata schema
+    shared_ptr<MetadataSchema> gpsSchema(new MetadataSchema(GPS_SCHEMA_NAME));
+
+    // Add description to the schema
+    gpsSchema->add(gpsDesc);
+
+    cout << "Adding metadata schema '" << GPS_SCHEMA_NAME << "'..." << endl;
+
+    // Add schema to metadata stream
+    mdStream.addSchema(gpsSchema);
+
+    shared_ptr<Metadata> gpsMetadata;
+
+    // Let there be an UFO moving around some point on Earth
+    const int nPoints = 32768;
+    for(int i = 0; i < nPoints; i++)
+    {
+        float lat =   37.235 + cos(i/25.0*2.0*M_PI) * 0.001;
+        float lng = -115.811 + sin(i/25.0*2.0*M_PI) * 0.001;
+        float time = i/25.0;
+        cout << "Adding metadata item 'lat " << lat << " lng " << lng << "'";
+        cout << " with associated time " << time << endl;
+
+        // Create a metadata item
+        gpsMetadata = shared_ptr<Metadata>(new Metadata(gpsDesc));
+
+        // Fill item fields
+        gpsMetadata->push_back(FieldValue(GPS_COORD_LAT_FIELD, lat));
+        gpsMetadata->push_back(FieldValue(GPS_COORD_LNG_FIELD, lng));
+        gpsMetadata->push_back(FieldValue(GPS_TIME_FIELD, time));
+
+        // Add to metadata a new item
+        mdStream.add(gpsMetadata);
+    }
+
+    cout << "Saving metadata..." << endl << endl;
+
+    // Save metadata to video file with custom compressor
+    shared_ptr<ICompressor> compressor = make_shared<MyLZWCompressor>();
+
+    // Register compressor before it could be used in saving and loading
+    vmf::registerCompressor(compressor);
```

Comment:
looks like registering a custom compressor is not necessary when it's used via pointer;
but probably we should allow using it via ID, like `mdStream.save("my_lzw");`
a?